<?xml version="1.0"?>
<launch>
    <include file="$(find amigo_machine_files)/machines/amigo1.machine" />
    <include file="$(find amigo_machine_files)/machines/amigo2.machine" />
    <include file="$(find amigo_machine_files)/machines/amigo3.machine" />
    <include file="$(find amigo_machine_files)/machines/amigo4.machine" />
    <include file="$(find amigo_machine_files)/machines/localhost.machine" />

    <arg name="machine" default="localhost"/>

    <node pkg="amigo_arm_navigation" type="grasp_precompute_action" name="grasp_precompute_left" output="screen" machine="$(arg machine)">
    	<remap from="/get_ik_solver_info" to="/amigo/left_arm_kinematics/get_ik_solver_info"/>
    	<remap from="/get_constraint_aware_ik" to="/amigo/left_arm_kinematics/get_constraint_aware_ik"/>
    	<remap from="/joint_measurements" to="/amigo/left_arm/measurements"/>
    	<remap from="/spindle_measurement" to="/amigo/torso/measurements" />
    	<remap from="/spindle_reference" to="/amigo/torso/references"/>
    	<param name="side" value="left"/>
        <param name="max_yaw_delta" value="2.0"/>
        <param name="yaw_sampling_step" value="0.05"/> <!-- [rad] -->
        <param name="spindle_sampling_step" value="0.05"/> <!-- [m] -->
        <param name="pre_grasp_delta" value="0.20"/>
        <param name="pre_grasp_inbetween_sampling_steps" value="20"/>
        <param name="spindle_min" value="0.1"/>
        <param name="spindle_max" value="0.4"/>
        <param name="tf_prefix" value="amigo"/>
    </node>
    
    <node pkg="amigo_arm_navigation" type="grasp_precompute_action" name="grasp_precompute_right" output="screen" machine="$(arg machine)">
    	<remap from="/get_ik_solver_info" to="/amigo/right_arm_kinematics/get_ik_solver_info"/>
    	<remap from="/get_constraint_aware_ik" to="/amigo/right_arm_kinematics/get_constraint_aware_ik"/>
    	<remap from="/joint_measurements" to="/amigo/right_arm/measurements"/>
    	<remap from="/spindle_measurement" to="/amigo/torso/measurements" />
    	<remap from="/spindle_reference" to="/amigo/torso/references"/>
    	<param name="side" value="right"/>
    	<param name="max_yaw_delta" value="2.0"/>
        <param name="yaw_sampling_step" value="0.05"/> <!-- [rad] -->
        <param name="spindle_sampling_step" value="0.05"/> <!-- [m] -->
        <param name="pre_grasp_delta" value="0.20"/>
        <param name="pre_grasp_inbetween_sampling_steps" value="20"/>
        <param name="spindle_min" value="0.1"/>
        <param name="spindle_max" value="0.4"/>
        <param name="tf_prefix" value="amigo"/>
    </node>

</launch>
